### DISCUSSION TITLE
[Guides Enhancement] Add Search using Algolia or similar

### DISCUSSION THREAD

**Discussion by Michoels** [rails/rails#52019] [Docs]
Can we add powerful search to the Rails guides?
I often need to find specific areas and am forced to use Google (alas).

Many documentations sites use Algolia's [Free for Open Source plan](https://www.algolia.com/for-open-source/).
Popular open source projects using Algolia for their docs:
* [Laravel](https://laravel.com/docs/11.x)
* [Tailwind CSS](https://tailwindcss.com/docs/installation)
* [React.js](https://react.dev/)
* [Vue.js](https://vuejs.org/guide/introduction.html)

While there are many open source, self-hosted, and custom alternatives, Algolia works extraordinarily well and (in theory) should be very straightforward to set up. 

While a custom solution (e.g. using [Lunr.js](https://github.com/olivernn/lunr.js)) would be tempting, it would take a tremendous effort to recreate Algolia's search quality ourselves. Why accept anything but the best?

The downside is that we'd be relying on a proprietary, closed-source solution.
That being said, many massive open source projects have been relying on it for many years.

I'd be willing to pitch in with this, but the Foundation would need to work things out with Algolia first.

The Algolia FAQ and contact form is [here](https://www.algolia.com/for-open-source/).

Thanks!

**Comment by Michoels** [comment]
Related:
Discussion https://github.com/rails/rails/discussions/51963

**Comment by AmandaPerino** [comment]
Hi @Michoels (and @madhums from #51963) - This is probably one of the most requested features for the docs, and so yes, we do have a Search bar on the wish list. 
Thank you for suggesting Algolia. I've added it to the list and we'll look into it when we are ready to start working on the search bar. I'll circle back to you @Michoels if this is the route we take. If so, we'd love your help getting it up and running.
